lib/pbench/server/__init__.py

```diff
@@ -236,25 +284,56 @@ def _get_valid_path(self, req_val, dir_val, logger):
                 if logger:
                     logger.error(
                         "The {} directory, does not resolve to a directory ('{}')",
-                        req_val,
+                        env_name,
                         dir_path,
                     )
                 return None
         return dir_path
 
-    def get_valid_dir_option(self, req_val, dir_val, logger):
-        """get_valid_dir_option - get the directory path raising an Error
-        if it is not a directory, returning a Path directory object.
+    def get_valid_dir_option(
+        self, env_name: str, dir_val: str, logger: Union[None, Logger]
+    ) -> Optional[Path]:
+        """Get the validated directory option from the given section.
+
+        Args:
+            env_name : Legacy environment name to display in error messages
+            dir_val : A directory value to resolve to a real directory
+            logger : Logger to use when low-level error messages should be
+                emitted
+
+        Returns:
+            A Path directory object, None if the directory value does not
+                resolve to a real directory.
         """
-        dir_path = self._get_valid_path(req_val, dir_val, logger)
+        dir_path = self._get_valid_path(env_name, dir_val, logger)
         if not dir_path:
             return None
 
         return dir_path
 
-    def timestamp(self):
-        """
-        "Return the current timestamp formatted as a string of the following form:
-                  <YYYY>-<MM>-<DD>T<hh>:<mm>:<ss>-<TZ>
+    def timestamp(self) -> str:
+        """Generate a time stamp string.
+
+        Returns:
+            The current timestamp formatted as a string of the following form:
+                <YYYY>-<MM>-<DD>T<hh>:<mm>:<ss>-<TZ>
         """
         return tstos(_time())
+
+
+def get_pbench_server_config(cfg_name: str) -> PbenchServerConfig:
```

Comment:
Adding the validation asserts is fine. I think it would have been a bit neater to make this a class factory method, although it's not a big deal.

```
    @classmethod
    def get_config(cls: Class, cfg_name: str) -> PbenchServerConfig:
        sc = cls(cfg_name)
        req_props = [getattr(sc, attr) for attr in PbenchServerConfig.REQ_PROPS]
        assert len(req_props) == len(PbenchServerConfig.REQ_PROPS)
        return sc
```

Reply:
And maybe instead of `len(PbenchServerConfig.REQ_PROPS)` it can be `len(__class__.REQ_PROPS)`?

Reply:
Or, at least, `len(cls.REQ_PROPS)`....